### PR TITLE
No more are you sure boxes

### LINKED ISSUE #3883: Feature request: Close on multiple tabs confirmation dialog being optional.
When closing terminal with multiple tabs, it's always a confirmation dialog shows up, forcing you to confirm closing. It's so annoying.

Please make it optional, through profile settings.
Thank you.

### PR DESCRIPTION
**Pull request by rstat1**
## Summary of the Pull Request
So this PR adds a profile setting called "confirmCloseAllTabs", that allows one to enable or disable the "Do you want close all tabs?" dialog that appears when you close a window with multiple open tabs. It current defaults to "true". Also adds a checkbox to that dialog that also sets "confirmCloseAllTabs"

 
## PR Checklist
* [x] Closes #3883 
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [ ] Tests added/passed
* [ ] Requires documentation to be updated
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Detailed Description of the Pull Request / Additional comments
I added a checkbox to the close dialog to set this setting, but I'm not sure how to best go about actually changing the setting from code; am open to suggestions, as to how it should be done, or if I should also just remove it and stick with the profile setting.

## Validation Steps Performed
1. Set "confirmCloseAllTabs" to false in my profile.json file.
2. Opened a 2nd tab.
3. Closed the window
4. Observed that there was no confirmation before the window closed.
5. Set "confirmCloseAllTabs" to true
6. Repeat steps 2 and 3
7. Observe that there was a confirmation before the window closed.

### CODE REVIEW

File: src/cascadia/TerminalApp/userDefaults.json [hunk 1]
```diff
@@ -6,6 +6,8 @@
 
     "defaultProfile": "{61c54bbd-c2c6-5271-96e7-009a87ff44bf}",
 
+    "confirmCloseAllTabs": true,
```
Comment (zadjii-msft) [inline src/cascadia/TerminalApp/userDefaults.json]: I'd probably remove this from this file - this file is just the "blank" `profiles.json` file we create when the Terminal is first run, and is intended to remain as minimal as possible (hence why very few of the other settings are in it).

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 2]
```diff
@@ -920,9 +926,9 @@ namespace winrt::TerminalApp::implementation
     //   than one tab opened, show a warning dialog.
     void TerminalPage::CloseWindow()
     {
-        if (_tabs.size() > 1)
+        if (_tabs.size() > 1 && _settings->GlobalSettings().GetConfirmCloseAllTabs())
         {
-            _ShowCloseWarningDialog();
+            _ShowCloseWarningDialog();
```
Comment (zadjii-msft) [inline src/cascadia/TerminalApp/TerminalPage.cpp]: These trailing spaces are probably why the code formatting check failed 😉

File: src/cascadia/TerminalApp/userDefaults.json (hunk 1 above)
Comment (rstat1) [reply src/cascadia/TerminalApp/userDefaults.json]: Done.

File: src/cascadia/TerminalApp/TerminalPage.cpp (hunk 2 above)
Comment (rstat1) [reply src/cascadia/TerminalApp/TerminalPage.cpp]: Ah OK, I was wondering about that. I'm used to VSCode where trailing whitespace is cleaned up for me automatically.

**Comment by rstat1** [comment]
Ok, so I also removed the checkbox (and related event handlers) as well. Though I would like to revisit doing that later.

File: src/cascadia/TerminalApp/Resources/en-US/Resources.resw [hunk 3]
```diff
@@ -1,17 +1,17 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <root>
-  <!--
-    Microsoft ResX Schema
-
+  <!--
```
Comment (miniksa) [inline src/cascadia/TerminalApp/Resources/en-US/Resources.resw]: Please undo this file. Whitespace only changes.

File: src/cascadia/TerminalApp/userDefaults.json [hunk 4]
```diff
@@ -5,7 +5,7 @@
     "$schema": "https://aka.ms/terminal-profiles-schema",
 
     "defaultProfile": "{61c54bbd-c2c6-5271-96e7-009a87ff44bf}",
-
+
```
Comment (miniksa) [inline src/cascadia/TerminalApp/userDefaults.json]: Please revert this file. Whitespace only changes.

File: src/cascadia/TerminalApp/GlobalAppSettings.cpp [hunk 5]
```diff
@@ -1,467 +1,482 @@
-// Copyright (c) Microsoft Corporation.
```
Comment (carlos-zamora) [inline src/cascadia/TerminalApp/GlobalAppSettings.cpp]: Looks like this entire file wasn't actually changed either.

File: src/cascadia/TerminalApp/Resources/en-US/Resources.resw [hunk 6]
```diff
@@ -210,4 +210,4 @@ Temporarily using the Windows Terminal default settings.
   <data name="CloseWindowWarningTitle" xml:space="preserve">
     <value>Do you want to close all tabs?</value>
   </data>
-</root>
+</root>
```
Comment (carlos-zamora) [inline src/cascadia/TerminalApp/Resources/en-US/Resources.resw]: Same here. Revert changes to this file.

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 7]
```diff
@@ -243,6 +243,7 @@ namespace winrt::TerminalApp::implementation
         auto secondaryButtonText = RS_(L"Cancel");
 
         WUX::Controls::ContentDialog dialog;
+        WUX::Controls::CheckBox remember;
```
Comment (carlos-zamora) [inline src/cascadia/TerminalApp/TerminalPage.cpp]: This seems to not be used anywhere.

File: src/cascadia/TerminalApp/userDefaults.json [hunk 8]
```diff
@@ -30,4 +30,4 @@
     // Add any keybinding overrides to this array.
     // To unbind a default keybinding, set the command to "unbound"
     "keybindings": []
-}
+}
```
Comment (carlos-zamora) [inline src/cascadia/TerminalApp/userDefaults.json]: Undo changes to this file.

File: src/cascadia/TerminalApp/GlobalAppSettings.cpp (hunk 5 above)
Comment (rstat1) [reply src/cascadia/TerminalApp/GlobalAppSettings.cpp]: There was weirdness at play here. I was attempting to fix a merge conflict in this file (something about snapping to a grid)  which I did, but for some reason Github thought I changed the entire file.

File: src/cascadia/TerminalApp/Resources/en-US/Resources.resw (hunk 6 above)
Comment (rstat1) [reply src/cascadia/TerminalApp/Resources/en-US/Resources.resw]: Not sure what you want me to do here. Everywhere I've looked there doesn't appear to be anything to revert.

File: src/cascadia/TerminalApp/userDefaults.json (hunk 8 above)
Comment (rstat1) [reply src/cascadia/TerminalApp/userDefaults.json]: Done.

**Comment by cinnamon-msft** [comment]
I dig this :)

**Pull request merged**